### PR TITLE
JsSetCurrentContext parameter should be _In_opt_

### PR DESCRIPTION
**Pull request by akrantz**
Since JS_INVALID_REFERENCE (= NULL) is a valid input to JsSetCurrentContext, the parameter should be marked as _In_opt_.

### CODE REVIEW

**Comment by kfarnung** [comment]
@digitalinfinity @MSLaguana @liminzhu Any concerns about this change since it applied to ChakraCommon?  It's fixing the SAL annotation to be more permissive, so I think we should be ok, but I don't know if we need to run it past the Windows API board.

Other than that it LGTM, the old annotation was wrong.

**Comment by digitalinfinity** [comment]
Presumably needs a doc to be updated somewhere. @liminzhu do you know? And I'll defer to limin on the API board but my assumption would be no.

File: lib/Jsrt/ChakraCommon.h
Comment (digitalinfinity) [inline lib/Jsrt/ChakraCommon.h]: My SAL fu is not that great- what is the intent behind this change?

Comment (MSLaguana) [reply lib/Jsrt/ChakraCommon.h]: These annotations don't seem to work on non-windows builds; I believe that we have a header somewhere that `#define`s the annotations away to nothing, but it seems to be missing `_Post_`

Comment (akrantz) [reply lib/Jsrt/ChakraCommon.h]: Sorry, this isn't needed here and I didn't realize it was checked in. Let me double-check with this removed.

File: lib/Jsrt/ChakraCommon.h
Comment (kfarnung) [inline lib/Jsrt/ChakraCommon.h:1039]: You'll need to update the annotation in Jsrt.cpp:

https://github.com/Microsoft/ChakraCore/blob/6543dfb6a478d6a5f8aece6285ba12566c634797/lib/Jsrt/Jsrt.cpp#L793

Comment (akrantz) [reply lib/Jsrt/ChakraCommon.h:1039]: Thanks. I pushed a commit with the change.

**Comment by liminzhu** [comment]
I think not but let me run this by Dave Risney to be certain.

**Comment by liminzhu** [comment]
Forgot to post update here. Kyle and I consulted the winsdk folks - this is considered as API change but a simple one that only requires an email based review. We aren't in a rush to merge this in at the moment, so once we get the green light from winsdk folks I will post another update here.

File: lib/Jsrt/ChakraCommon.h
Comment (Adamskeeper) [reply lib/Jsrt/ChakraCommon.h]: how do I find someone to help me make my first app

Comment (Adamskeeper) [reply lib/Jsrt/ChakraCommon.h]: i have full detailed information on my app phase 1 phase2 and phase 3 I need someone with a kind heart who wants to help

**Review by rhuanjl — APPROVED** [review]
This has sat around far too long; it's a sensible change, so going to merge it once CI completes

EDIT: ugh CI failed due to running out of disk space - may postpone merging this until repository migration is done

**Comment by rhuanjl** [comment]
Skipped the CI rerun and merged this for efficiency - it had passed pre-migration apart from windows 7 and was such a simple change.

**Pull request merged**
